### PR TITLE
feat(slo): support folder field

### LINKED ISSUE #1527: [Feature Request]: Allow defining Folder for SLOs
### Feature Request

Could the provider be updated to support defining the Folder the SLO dashboard will live in? I see it's currently available in the API Spec https://grafana.com/docs/grafana-cloud/alerting-and-irm/slo/api/slo-api.yaml

Thanks!

### PR DESCRIPTION
**Pull request by inikotoran**
### Summary

In order for users to be able to define folder for SLO resource, we need to upgrade github.com/grafana/slo-openapi-client to latest version. This mainly require changing the usage of `Slo` struct to `SloV00Slo`. The new struct is generated by OpenAPI spec and since the API request function expect the versioned struct, it would be easier to use the new struct directly.

Related to the folder input, the new field is called `folder_uid` and it's optional.

#### Local

The following tests has been conducted locally and applied in Grafana Cloud Enterprise account.
- [x] Can create SLO without defining `folder_uid`
- [x] Can create SLO with `folder_uid`
- [x] Throws error when folder UID is invalid

### CODE REVIEW

File: internal/resources/slo/resource_slo.go [hunk 1]
```diff
@@ -51,6 +51,12 @@ Resource manages Grafana SLOs.
 				Description:  `Description is a free-text field that can provide more context to an SLO.`,
 				ValidateFunc: validation.StringLenBetween(0, 1024),
 			},
+			"folder": {
```
Comment (julienduchesne) [inline internal/resources/slo/resource_slo.go]:
```suggestion
			"folder_uid": {
```

Can we call it that way to make it clear that we expect the `grafana_folder.<>.uid` attribute?

Comment (inikotoran) [reply internal/resources/slo/resource_slo.go]: hi @julienduchesne thank your for your review. It make sense, I've updated it here: 7bb5292da7c52d4fbef79a4f1359ef7dadb48ad3

Comment (inikotoran) [reply internal/resources/slo/resource_slo.go]: going to fix the test locally, edit: fixed here: ea5d82571ede312142a444b7686784c8fa967098

Comment (inikotoran) [reply internal/resources/slo/resource_slo.go]: @julienduchesne is it mandatory to add tests for the `folder_uid` input? If that's the case, I'm thinking to follow [resource_dashboard_test](https://github.com/grafana/terraform-provider-grafana/blob/main/internal/resources/grafana/resource_dashboard_test.go#L316-L336) implementation by creating folder first

**Comment by julienduchesne** [comment]
👋 For some reason, the tests don't pass anymore when these changes are applied. I will investigate more tomorrow 👀

**Comment by julienduchesne** [comment]
LGTM! Thanks!

**Review by julienduchesne — APPROVED** [review]

**Pull request merged**